### ISSUE TITLE
Visualization and understanding of latent space

### ISSUE THREAD

**Issue by ChenFengYe** [ChenFengYe/motion-latent-diffusion#12]
Related to #7. From [mmdrahmani](https://github.com/mmdrahmani),

> I also have another basic question. I would like to understand the latent dimension of VAE. I'd like to know what the model has learned. Essentially, I am assuming if we could visualize the latent dimension, different actions would be clustered in different locations of the latent space. For example see the figure attached, for my analysis on the latent dimension of a simple VAE using mnist data. As you can see, the 10 digits are clearly clustered. I hope this kind of analysis is possible with mld-vae. (maybe I should open a new issue?)
![image](https://user-images.githubusercontent.com/16475892/222059384-32caf25b-52bf-4d16-befc-e6d4ff6802c9.png)

**Comment by ChenFengYe** [comment]
We think motion VAE could extract the motion principal features, just like PCA does.
Thus, we try to visualize the motion latent space like below. We will update this part to our arxiv paper and camera-ready version soon.

From left to right, it shows the evolved latent codes during the inference of diffusion models. However, this is only an intuitive figure of the visualization of latent diffusion (only 30 motions). We also try to visualize much more motion results (like 300 or 3000). Unfortunately, the classification is not that obvious.

![image](https://user-images.githubusercontent.com/16475892/222061148-895d4ed5-68a2-407c-9442-777420880a14.png)

**Comment by ChenFengYe** [comment]
Of course, we will upload the source code for this visualization soon.

**Comment by mmdrahmani** [comment]
Thanks a lot!

**Comment by ChenFengYe** [comment]
Hi mmdrahmani, we upload [the script](https://github.com/ChenFengYe/motion-latent-diffusion/blob/main/scripts/tsne.py) for this latent visualization. You can find the detail in FAQ of GitHub page.
